flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -1215,17 +1238,22 @@ private void jobMasterFailed(JobID jobId, Throwable cause) {
                 getMainThreadExecutor());
     }
 
-    CompletableFuture<Void> getJobTerminationFuture(JobID jobId) {
+    private CompletableFuture<Void> getTerminatedJobTerminationFuture(JobID jobId) {
```

Comment:
hm, here I'm not certain about the naming. It sounds odd. But I cannot come up with something else. The only thing I have in mind is `getJobTerminationFutureOrFailedFutureForRunningJob`. But I'm not happy about that one, either 🤔

Reply:
Sounds fine with me as well. 👍 It's just that the method itself doesn't throw the `DispatcherException` but returns a exceptionally completed future.